scripts/lib/CIME/case_st_archive.py

```diff
@@ -20,45 +20,143 @@ def _get_archive_file_fn(copy_only):
     """
     return shutil.copyfile if copy_only else shutil.move
 
-
 ###############################################################################
-def _get_datenames(case, last_date=None):
+def _get_datenames(rundir, casename):
 ###############################################################################
     """
-    Returns a list of datenames giving the dates of cpl restart files
-
-    If there are no cpl restart files, this will return []
+    Returns the datetime objects specifying the times of each file
+    Note we are assuming that the coupler restart files exist and are consistent with other component datenames
+    Not doc-testable due to filesystem dependence
     """
-
-    if last_date is not None:
-        try:
-            last = datetime.datetime.strptime(last_date, '%Y-%m-%d')
-        except ValueError:
-            expect(False, 'Could not parse the last date to archive')
-    logger.debug('In get_datename...')
-    rundir = case.get_value('RUNDIR')
+    logger.info(rundir)
```

Comment:
If you want to leave this info statement in please make it a complete sentence.

Reply:
I'll remove it and change the other one

Reply:
Done